lib/training_progress_manager.rb

```diff
@@ -74,6 +74,10 @@ def module_progress
     module_completed? ? I18n.t('training_status.completed') : completing
   end
 
+  def completion_time
+    return @tmu.completed_at - @tmu.created_at if completion_date
```

Comment:
This will be slightly clearer if you use `module_completed?`

I recommend doing it in two lines with a guard clause and then the calculation. That makes it a little more clear what the possible return values are (nil, or a duration).

Reply:
Done! @ragesoss Please have a look.